src/components/rules-creation-page/steps/IncludeExcludeSelectionStep.tsx

```diff
@@ -0,0 +1,236 @@
+        );
+    };
+
+    const changeModelName = (event: any) => {
+        setSelectedType(event.target.value);
+    };
+
+    const changeInclude = (includeRules: any) => {
+        const type: string = selectedType || "";
+
+        const oldIncludeRules = getIncludeRules();
+        const oldExcludeRules = getExcludeRules();
+
+        const ruleIndexesToExclude = _.difference(oldIncludeRules, includeRules).map(rule =>
+            oldIncludeRules.indexOf(rule)
+        );
+        const ruleIndexesToInclude = _.difference(includeRules, oldIncludeRules).map(rule =>
+            oldExcludeRules.indexOf(rule)
+        );
+
+        if (ruleIndexesToInclude.length > 0) {
+            onChange(syncRule.moveRuleFromExcludeToInclude(type, ruleIndexesToInclude));
+        } else if (ruleIndexesToExclude.length > 0) {
+            onChange(syncRule.moveRuleFromIncludeToExclude(type, ruleIndexesToExclude));
+        }
+    };
+
+    const getAllRules = () => {
+        const allRules =
+            selectedType && syncRule.metadataExcludeIncludeRules
+                ? [
+                      ...syncRule.metadataExcludeIncludeRules[selectedType].excludeRules,
+                      ...syncRule.metadataExcludeIncludeRules[selectedType].includeRules,
+                  ]
+                : [];
+
+        return allRules;
+    };
+
+    const getIncludeRules = () => {
+        return selectedType && syncRule.metadataExcludeIncludeRules
+            ? syncRule.metadataExcludeIncludeRules[selectedType].includeRules
+            : [];
+    };
+
+    const getExcludeRules = () => {
+        return selectedType && syncRule.metadataExcludeIncludeRules
+            ? syncRule.metadataExcludeIncludeRules[selectedType].excludeRules
+            : [];
+    };
+
+    const getRuleOptions = () => {
+        const allRules = getAllRules();
+
+        return allRules.map(rule => ({
+            value: rule,
+            text: includeExcludeRulesFriendlyNames[rule]
+                ? includeExcludeRulesFriendlyNames[rule]
+                : rule,
+        }));
+    };
+
+    return (
+        <React.Fragment>
+            <Toggle
+                label={i18n.t("Use default configuration")}
+                value={syncRule.useDefaultIncludeExclude}
+                onValueChange={changeUseDefaultIncludeExclude}
+            />
+            {!syncRule.useDefaultIncludeExclude && (
+                <div
+                    style={{
+                        display: "flex",
+                        flexDirection: "column",
+                        alignItems: "flex-start",
+                        marginTop: "20px",
+                    }}
+                >
+                    <Dropdown
+                        key={"model-selection"}
+                        items={modelSelectItems}
+                        onChange={changeModelName}
+                        value={selectedType ? selectedType : ""}
```

Comment:
If we simplify ``selectedType`` to be only string, this could be: ``value={selectedType}``.

Reply:
Done!